python_modules/dagster/dagster/_core/definitions/data_version.py

```diff
@@ -458,3 +450,14 @@ def _is_partitioned_or_downstream(self, *, key: AssetKey) -> bool:
                 self._is_partitioned_or_downstream(key=dep_key)
                 for dep_key in self.asset_graph.get_parents(key)
             )
+
+    # Volatility means that an asset is assumed to be constantly changing. We assume that observable
+    # source assets are non-volatile, since the primary purpose of the observation function is to
+    # determine if a source asset has changed. We assume that regular assets are volatile if they
+    # are at the root of the graph (have no dependencies).
+    @cached_method
```

Comment:
Originally, I was thinking (incorrectly) that this volatility thing would be recursively defined, but that's not the case. So maybe it makes more sense for this to just live on the asset graph? The operation is cheap so I don't see any benefit to it being cached